### PR TITLE
Refactoring beacon

### PR DESCRIPTION
**Pull request by bsmithgall**
- Remove the majority of the crufty code in opportunities/util and migrate it to various classes
- Add more class methods/form methods to further shrink the views
- Still TODO: write some unit tests for the new class methods

### CODE REVIEW

**Comment by bsmithgall** [comment]
@hamhands can you take a look at this? I'd like to get it merged as it will make working on Beacon a bit easier in the longer term.

File: purchasing/opportunities/forms.py [hunk 1]
```diff
@@ -72,11 +94,73 @@ def validate_phone_number(form, field):
         if len(value) != 10 and len(value) != 0:
             raise ValidationError('Invalid 10-digit phone number!')
 
-class SignupForm(Form):
+class CategoryForm(Form):
```
Comment (hamhands) [inline purchasing/opportunities/forms.py:136]: Nice. This is a good solution :+1:

File: purchasing/opportunities/util.py [hunk 2]
```diff
@@ -1,297 +1,16 @@
 # -*- coding: utf-8 -*-
 
-import datetime
```
Comment (hamhands) [inline purchasing/opportunities/util.py:3]: So most/all of this was moved to the forms.py?

**Comment by hamhands** [comment]
Cool, LGTM :+1:

**Pull request merged**